sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1044,6 +1044,34 @@ class Analyzer(
           failOnOuterReference(p)
           p
       }
+
+      // SPARK-17348
+      // Looking for a potential incorrect result case.
+      // When a correlated predicate is a non-equality predicate
+      // it must be placed at the immediate child operator.
+      // Otherwise, the pull up of the correlated predicate
+      // will generate a plan with a different semantics
+      // which could return incorrect result.
+      var continue : Boolean = true
```

Comment:
That scenario above is banned with this PR (and added as a new, negative, test case). However, the PR has a greater scope, Window function is another operator, we cannot pull up any correlated predicates, such as the example below:

    // Case 2: window func
    Seq((2,1)).toDF("c1","c2").createOrReplaceTempView("t1")
    Seq((1,1),(2,2)).toDF("c1","c2").createOrReplaceTempView("t2")
    sql("select c1 from t1 where c1 in (select max(t2.c1) over () from t2 where t1.c2 >= t2.c2)").show